tests/clone_test.go

```diff
@@ -874,3 +729,149 @@ func createSourceVM(options ...libvmi.Option) (*virtv1.VirtualMachine, error) {
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+	return snapshot
+}
+
+func waitSnapshotReady(snapshotName, snapshotNamespace string) *snapshotv1.VirtualMachineSnapshot {
+	var snapshot *snapshotv1.VirtualMachineSnapshot
+
+	EventuallyWithOffset(1, func() bool {
+		var err error
+		snapshot, err = kubevirt.Client().VirtualMachineSnapshot(snapshotNamespace).Get(context.Background(), snapshotName, v1.GetOptions{})
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		return snapshot.Status != nil && snapshot.Status.ReadyToUse != nil && *snapshot.Status.ReadyToUse
+	}, 180*time.Second, time.Second).Should(BeTrue(), "snapshot should be ready")
+
+	return snapshot
+}
+
+func waitSnapshotContentsExist(snapshotName, snapshotNamespace string) *snapshotv1.VirtualMachineSnapshot {
+	var contentsName string
+	var snapshot *snapshotv1.VirtualMachineSnapshot
+	EventuallyWithOffset(1, func() error {
+		var err error
+		snapshot, err = kubevirt.Client().VirtualMachineSnapshot(snapshotNamespace).Get(context.Background(), snapshotName, v1.GetOptions{})
+		ExpectWithOffset(2, err).ToNot(HaveOccurred())
+		if snapshot.Status == nil {
+			return fmt.Errorf("snapshot's status is nil")
+		}
+
+		if snapshot.Status.VirtualMachineSnapshotContentName != nil {
+			contentsName = *snapshot.Status.VirtualMachineSnapshotContentName
+		} else {
+			return fmt.Errorf("vm snapshot contents name is nil")
+		}
+
+		return nil
+	}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+
+	EventuallyWithOffset(1, func() error {
+		_, err := kubevirt.Client().VirtualMachineSnapshotContent(snapshotNamespace).Get(context.Background(), contentsName, v1.GetOptions{})
+		return err
+	}).ShouldNot(HaveOccurred())
+
+	return snapshot
+}
+
+func generateCloneFromVMWithParams(sourceVMName, sourceVMNamespace, targetVMName string) *clone.VirtualMachineClone {
+	vmClone := kubecli.NewMinimalCloneWithNS("testclone", sourceVMNamespace)
+
+	cloneSourceRef := &k8sv1.TypedLocalObjectReference{
+		APIGroup: pointer.P(vmAPIGroup),
+		Kind:     "VirtualMachine",
+		Name:     sourceVMName,
+	}
+
+	cloneTargetRef := cloneSourceRef.DeepCopy()
+	cloneTargetRef.Name = targetVMName
+
+	vmClone.Spec.Source = cloneSourceRef
+	vmClone.Spec.Target = cloneTargetRef
+
+	return vmClone
+}
+
+func generateCloneFromSnapshot(snapshotName, snapshotNamespace, targetVMName string) *clone.VirtualMachineClone {
+	vmClone := kubecli.NewMinimalCloneWithNS("testclone", snapshotNamespace)
+
+	cloneSourceRef := &k8sv1.TypedLocalObjectReference{
+		APIGroup: pointer.P(virtsnapshot.GroupName),
+		Kind:     "VirtualMachineSnapshot",
+		Name:     snapshotName,
+	}
+
+	cloneTargetRef := &k8sv1.TypedLocalObjectReference{
+		APIGroup: pointer.P(vmAPIGroup),
+		Kind:     "VirtualMachine",
+		Name:     targetVMName,
+	}
+
+	vmClone.Spec.Source = cloneSourceRef
+	vmClone.Spec.Target = cloneTargetRef
+
+	return vmClone
+}
+
+func createCloneAndWaitForFinish(vmClone *clone.VirtualMachineClone) {
```

Comment:
Pass name/namespace of the clone instead. I don't like that this is modifying the pointer that was passed in.

Reply:
>Pass name/namespace of the clone instead

We can't , the object is used to be created. 

> I don't like that this is modifying the pointer that was passed in.

That's indeed not needed. will change.

Reply:
Might make sense to return the created clone? (possibly in a follow up and only if it is ever used again)